.github/workflows/build-master.yml

```diff
@@ -0,0 +1,27 @@
+on:
+  push:
+    branches:
+      - master
+      # gh-mster-3 must be removed before merging this PR. It was added to test the pipeline.
+      - gh-mster-s3
+
+jobs:
+  build-static-page:
+    name: build
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_BUILDKIT: '1'
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    steps:
+      - name: print docker info
+        run: docker version && docker info
+      - uses: actions/checkout@v2
+      - name: build current docs
+        run: docker build --build-arg=ENABLE_ARCHIVES=true --target=deploy-source --output=./_site .
+      - name: list _site
+        run: ls -l _site
+      - name: upload files to S3 bucket
+        run: aws s3 sync --acl public-read _site s3://docs.docker.com-test-us-east-1/ --delete
```

Comment:
I'm wondering how well it picks up that the "archives" never change between builds; is that something we need to optimise? (e.g. resetting date/time if it compares that metadata)

Reply:
I ran this command twice. For the first run the S3 bucket was completely empty, and afterwards no index.html or any other file at top level were uploaded. In sub directories I could find index.html files. 🤔 

I ran it again with the ls -l _site commit to "debug" the GH action pipeline, now all files seems to be uploaded.
I wonder how predictable the aws s3 sync command is. @tiborvass any insights? :-)

Reply:
Alright, I found my problem why index.html was missing. I did more runs here and the upload/sync looks good now. Pages can be viewed after we fixed a configuration in AWS.
The version selection also works to see older docs.

Reply:
It could be that all files get synced due to new timestamps from the newly built container image for each pipeline run.
From a costs perspective it's inbound traffic from the internet and that's free.